components/FavoriteButton.tsx

```diff
@@ -0,0 +1,46 @@
+import FilledHeart from '@/public/icons/ic_fluent_heart_24_filled.svg';
+import styled, { css } from 'styled-components';
+
+export default function FavoriteButton({
+  $isDetailPage,
+  onToggleFavorite,
+  isFavorite,
+}) {
```

Comment:
The types for your props as well as whether they are mandatory or not are being inferred.
This does not work the way you want, if some props are optional, like `$isDetailPage`:

```suggestion
export default function FavoriteButton({
  $isDetailPage,
  onToggleFavorite,
  isFavorite,
}: {
  $isDetailPage?: boolean;
  onToggleFavorite: () => {};
  isFavorite: boolean;
}) {
```